src/clib/pioc_support.c

```diff
@@ -2931,13 +2968,20 @@ int PIOc_openfile_retry(int iosysid, int *ncidp, int *iotype, const char *filena
 #endif
 
         default:
-            free(file);
             {
                 char avail_iotypes[PIO_MAX_NAME + 1];
+
+                spio_ltimer_stop(ios->io_fstats->rd_timer_name);
+                spio_ltimer_stop(ios->io_fstats->tot_timer_name);
+                spio_ltimer_stop(file->io_fstats->rd_timer_name);
+                spio_ltimer_stop(file->io_fstats->tot_timer_name);
+
                 PIO_get_avail_iotypes(avail_iotypes, PIO_MAX_NAME);
                 return pio_err(ios, NULL, PIO_EBADIOTYPE, __FILE__, __LINE__,
                                 "Opening file (%s) failed. Invalid iotype (%s:%d) specified. Available iotypes are : %s", filename, pio_iotype_to_string(file->iotype), file->iotype, avail_iotypes);
             }
+            free(file->io_fstats);
```

Comment:
We might move the two free() calls before return pio_err() is called.
Also, before file is freed, we can use a temp variable to save file->iotype,
which will be used later in pio_err() call.

Reply:
Thanks. I will fix that soon.